.github/local-actions/branch-manager/lib/main.ts

```diff
@@ -0,0 +1,160 @@
+  core.info(JSON.stringify(pullRequest));
+  /** Whether any fatal validation failures were discovered. */
+  let hasFatalFailures = false;
+  /** The status information to be pushed as a status to the pull request. */
+  let statusInfo: CommmitStatus = await (async () => {
+    // Log validation failures and check for any fatal failures.
+    if (pullRequest.validationFailures.length !== 0) {
+      core.info(`Found ${pullRequest.validationFailures.length} failing validation(s)`);
+      for (const failure of pullRequest.validationFailures) {
+        hasFatalFailures = !failure.canBeForceIgnored || hasFatalFailures;
+        await core.group('Validation failures', async () => {
+          core.info(failure.message);
+        });
+      }
+    }
+
+    // With any fatal failure the check is not necessary to do.
+    if (hasFatalFailures) {
+      core.info('One of the validations was fatal, setting the status as pending for the pr');
+      return {
+        description: 'Waiting to check mergeability due to failing status(es)',
+        state: 'pending',
+      };
+    }
+
+    try {
+      git.run(['checkout', 'main']);
+      /**
+       * A merge strategy used to perform the merge check.
+       * Any concrete class implementing MergeStrategy is sufficient as all of our usage is
+       * defined in the abstract base class.
+       * */
+      const strategy = new AutosquashMergeStrategy(git);
+      await strategy.prepare(pullRequest);
+      await strategy.check(pullRequest);
+      core.info('Merge check passes, setting a passing status on the pr');
+      return {
+        description: `Merges cleanly to ${pullRequest.targetBranches.join(', ')}`,
+        state: 'success',
+      };
+    } catch (e) {
+      // As the merge strategy class will express the failures during checks, any thrown error is a
+      // failure for our merge check.
+      let description: string;
+      if (e instanceof MergeConflictsFatalError) {
+        core.info('Merge conflict found');
+        description = `Unable to merge into ${e.failedBranches.join(
+          ', ',
+        )} please update changes or PR target`;
+      } else {
+        core.info('Unknown error found when checking merge');
+        description =
+          'Cannot cleanly merge to all target branches, please update changes or PR target';
+      }
+      return {
+        description,
+        state: 'failure',
+      };
+    }
+  })();
+
+  await git.github.repos.createCommitStatus({
+    ...repo,
+    ...statusInfo,
+    sha: pullRequest.headSha,
+    context: 'Branch Manager',
+  });
+}
+
+/** The token for the angular robot to perform actions. */
+const token = await getAuthTokenFor(ANGULAR_ROBOT, true);
+/** The repository name for the pull request. */
+const repo = core.getInput('repo', {required: true, trimWhitespace: true});
+/** The owner of the repository for the pull request. */
+const owner = core.getInput('owner', {required: true, trimWhitespace: true});
```

Comment:
Should we just retrieve owner/name of the repo through context? Like it's also done for the `getAuthTokenFor`. Right now we have this setting but it's not used consistently- and not sure if needed

Reply:
Because this always is running in the dev-infra repo, the context is not the "correct" repo and owner